subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c

```diff
@@ -132,6 +138,94 @@ static void azure_iot_hub_notify_event(struct azure_iot_hub_evt *evt)
 	}
 }
 
+const char *state_name_get(enum connection_state state)
+{
+	switch (state) {
+	case STATE_IDLE: return "STATE_IDLE";
+	case STATE_INIT: return "STATE_INIT";
+	case STATE_CONNECTING: return "STATE_CONNECTING";
+	case STATE_CONNECTED: return "STATE_CONNECTED";
+	case STATE_DISCONNECTING: return "STATE_DISCONNECTING";
+	default: return "STATE_UNKNOWN";
+	}
+}
+
+static void connection_state_set(enum connection_state new_state)
+{
+	bool notify_error = false;
+
+	/* Check for legal state transitions. */
+	switch (connection_state) {
+	case STATE_IDLE:
+		if (new_state == STATE_INIT) {
+			/* Legal state change. */
+			break;
+		}
+		notify_error = true;
+		break;
```

Comment:
Sure. That works. I originally had it this way, but changed it because it might look a little bit confusing at first glace. Especially for the if statements checking multiple ```new_state``` values.